packages/hoprd/src/admin.ts

```diff
@@ -38,6 +38,19 @@ export class AdminServer {
       return false
     }
 
+    // Other clients different to `hoprd` might pass the `apiToken` via a
+    // query param since they won't be on the same domain the node is hosted,
+    // and thus, unable to set the `apiToken` via cookies. Using `req.url` we
+    // can detect these cases and provide the ability for any client that
+    // knows the `apiToken` to reach your HOPR node.
+    if (req.url) {
+      const query = parse(req.url).query
+      const [_, apiToken] = (query && query.split('=')) || []
```

Comment:
Does this assume a single k/v pair in the url's query string? Wouldn't the match fail if there were more?

Reply:
Indeed, this is an issue for sure. Let me see if I can do something a bit better.

Reply:
Updated with the `URL` modern API, which allows specific targeting for search parameters instead of split magic 🌟